themes/finna2/js/finna-common.js

```diff
@@ -81,54 +81,56 @@ finna.common = (function finnaCommon() {
     VuFind.listen('results-loaded', () => {
       initResultScripts(document.querySelector('.js-result-list'), false);
     });
-
+    const dropdownMappings = {
+      sort: {
+        selector: '.search-controls form.search-sort select',
+        label: 'Sort',
+        sessionSelector: 'sort-dropdown a.dropdown-toggle',
+      },
+      limit: {
+        selector: '.search-controls form.search-result-limit select',
+        label: 'Results per page',
+        sessionSelector: 'limit-dropdown a.dropdown-toggle',
+      }
+    };
     // Set up Finna's dropdown-based sort and limit controls:
-    document.querySelectorAll('.search-controls .sort-option-container .dropdown-menu a, .search-controls .limit-option-container .dropdown-menu a').forEach(link => {
-      if (link.dataset.ajaxPagination) {
+    document.querySelectorAll('.search-controls .sort-option-container .dropdown, .search-controls .limit-option-container .dropdown').forEach(dropdown => {
```

Comment:
Tuntuu hassulta säilyttää nämä täällä, kun on tuo dropdownMappings. Mitä jos kääntäisi tämän niin päin, että nämä selektorit olisivat myös mappauksessa, ja käytäisiin mappaukset läpi loopissa?